pkg/codegen/docs/gen_creation_example_for_csharp.go

```diff
@@ -0,0 +1,269 @@
+		rootNamespace := csharpNamespaceName(namespaces, pkg)
+
+		namespace := csharpNamespaceName(namespaces, module)
+		if strings.ToLower(namespace) == "index" {
+			namespace = ""
+		}
+		namespaceTokens := strings.Split(namespace, "/")
+		for i, name := range namespaceTokens {
+			namespaceTokens[i] = title(name, "csharp")
+		}
+		namespace = strings.Join(namespaceTokens, ".")
+
+		if namespace != "" {
+			namespace = "." + namespace
+		}
+
+		qualifiedMemberName := fmt.Sprintf("%s%s.%s", rootNamespace, namespace, title(member, "csharp"))
+		return qualifiedMemberName
+	}
+
+	indentSize := 0
+	buffer := bytes.Buffer{}
+	write := func(format string, args ...interface{}) {
+		buffer.WriteString(fmt.Sprintf(format, args...))
+	}
+
+	indent := func() {
+		buffer.WriteString(strings.Repeat(" ", indentSize))
+	}
+
+	indended := func(f func()) {
+		indentSize += 2
+		f()
+		indentSize -= 2
+	}
+
+	seenTypes := codegen.NewStringSet()
+	var writeValue func(valueType schema.Type)
+	writeValue = func(valueType schema.Type) {
+		switch valueType {
+		case schema.BoolType:
+			write("true|false")
+		case schema.IntType:
+			write("0")
+		case schema.NumberType:
+			write("0.0")
+		case schema.StringType:
+			write("\"string\"")
+		case schema.ArchiveType:
+			write("new FileAsset(\"./file.txt\")")
+		case schema.AssetType:
+			write("new StringAsset(\"Hello, world!\")")
+		}
+
+		switch valueType := valueType.(type) {
+		case *schema.ArrayType:
+			write("new []\n")
+			indent()
+			write("{\n")
+			indended(func() {
+				indent()
+				writeValue(valueType.ElementType)
+				write("\n")
+			})
+			indent()
+			write("}")
+		case *schema.MapType:
+			write("{\n")
+			indended(func() {
+				indent()
+				write("[\"string\"] = ")
+				writeValue(valueType.ElementType)
+				write("\n")
+			})
+			indent()
+			write("}")
+		case *schema.ObjectType:
+			if seenTypes.Has(valueType.Token) && objectTypeHasRecursiveReference(valueType) {
+				write("type(%s)", valueType.Token)
+				return
+			}
+
+			seenTypes.Add(valueType.Token)
+			typeName := argumentTypeName(valueType)
+			write("new %s\n", typeName)
+			indent()
+			write("{\n")
+			overrides := csharpPropertyNameOverrides(valueType.Properties)
+			indended(func() {
+				for _, p := range valueType.Properties {
+					indent()
+					write("%s = ", resolveCSharpPropertyName(p.Name, overrides))
+					writeValue(p.Type)
+					write(",\n")
+				}
+			})
+			indent()
+			write("}")
+		case *schema.ResourceType:
+			write("reference(%s)", valueType.Token)
+		case *schema.EnumType:
+			cases := make([]string, 0)
+			for _, c := range valueType.Elements {
+				if stringCase, ok := c.Value.(string); ok && stringCase != "" {
+					cases = append(cases, fmt.Sprintf("%q", stringCase))
+				} else if intCase, ok := c.Value.(int); ok {
+					cases = append(cases, strconv.Itoa(intCase))
+				} else {
+					if c.Name != "" {
+						cases = append(cases, c.Name)
+					}
+				}
+			}
+
+			write(strings.Join(cases, "|"))
```

Comment:
For this and all other languages, I think we want to ensure the output is a legal string value so that the copy/pasted code block parses correctly - so would put the `|` inside the quotes, not outside.

That is instead of:

```
log_type="LOG_TYPE_UNSPECIFIED"|"ADMIN_READ"|"DATA_WRITE"|"DATA_READ",
```

Emit:

```
log_type="LOG_TYPE_UNSPECIFIED|ADMIN_READ|DATA_WRITE|DATA_READ",
```

Reply:
Done ✅